packages/native/src/components/basic/TextField.js

```diff
@@ -83,19 +83,20 @@ class TextField extends Component {
 		const performUpdate = () => {
 			this.setState({
 				currentValue: value,
+			}, () => {
```

Comment:
Did we miss the `setState` callback earlier?

Reply:
yep, good find!

Reply:
But the code should be:

```
const performUpdate = () => {
	this.setState({
		currentValue: value,
	}, () => {
		if (isDefaultValue) {
			this.updateQuery(value, props);
			// [handle onValueChange here]
		} else {
			// debounce for handling text while typing
			this.handleTextChange(value); // [handle onValueChange inside handleTextChange]
		}
		this.locked = false;
	});
};
```